File: python/kocab/status.py
UNDEFINED_ID = 0
UNDEFINED_NAME = "undefined"
DEFAULT_ID = 1
DEFAULT_NAME = "default"
RUN_ID = 2
RUN_NAME = "run"
STOP_ID = 8
STOP_NAME = "stop"


class Status:
    def __init__(self, id: int, name: str):
        self.id = id
        self.name = name

    def __str__(self):
        return self.name


def get_status_by_id(id: int) -> Status:
    if id == 1:
        return Status(DEFAULT_ID, DEFAULT_NAME)
    elif id == 2:
        return Status(RUN_ID, RUN_NAME)
    elif id == 8:
        return Status(STOP_ID, STOP_NAME)
    else:
        return Status(UNDEFINED_ID, UNDEFINED_NAME)


def parse_raw_status(raw_id: str) -> Status:
    return get_status_by_id(int(raw_id))

File: app/timers/status/status.go
package status

import "strconv"

const (
	UndefinedID   = int(0)
	UndefinedName = string("undefined")
	DefaultID     = int(1)
	DefaultName   = string("default")
	RunID         = int(2)
	RunName       = string("run")
	StopID        = int(8)
	StopName      = string("stop")
)

type Status struct {
	Id   int    `json:"id"`
	Name string `json:"name"`
}

func New() Status {
	return newStatus(UndefinedID, UndefinedName)
}

func newStatus(id int, name string) Status {
	return Status{
		Id:   id,
		Name: name,
	}
}

func GetByID(id int) Status {
	switch id {
	case DefaultID:
		return newStatus(DefaultID, DefaultName)
	case RunID:
		return newStatus(RunID, RunName)
	case StopID:
		return newStatus(StopID, StopName)
	default:
		return newStatus(UndefinedID, UndefinedName)
	}
}

func ParseRaw(raw string) (Status, error) {
	id, err := strconv.Atoi(raw)
	if err != nil {
		return Status{}, err
	}

	return GetByID(id), nil
}
